src/dbally/similarity/elastic_store.py

```diff
@@ -0,0 +1,132 @@
+from typing import Dict, List, Optional, Tuple
+
+import numpy as np
+from elasticsearch import AsyncElasticsearch
+from elasticsearch.helpers import async_bulk
+
+from dbally.embedding_client.base import EmbeddingClient
+from dbally.similarity.store import SimilarityStore
+
+
+class ElasticStore(SimilarityStore):
+    """
+    The ElasticStore class stores text embeddings using knnSearch.
+    """
+
+    def __init__(
+        self,
+        index_name: str,
+        embedding_client: EmbeddingClient,
+        host: str,
+        http_auth_tuple: Tuple[str, str],
+        ca_cert_path: str,
+        search_algorith: Optional[Dict] = None,
+    ) -> None:
+        """
+        Initializes the ElasticStore.
+
+        Args:
+            index_name: The name of the index.
+            embedding_client: The client to use for creating text embeddings.
+
+        """
+        super().__init__()
+        self.es = AsyncElasticsearch(
+            hosts=host,
+            http_auth=http_auth_tuple,
+            ca_certs=ca_cert_path,
+        )
+        self.index_name = index_name
+        self.embedding_client = embedding_client
+        self.indices = []
+        self.search_algorithm = search_algorith or {
+            "knn": {
+                "field": "search_vector",
+                "k": 10,
+                "num_candidates": 50,
+            }
+        }
+
+    async def generate_data(self, data):
```

Comment:
This method `generate_data()` seems unnecessary to me. It looks quite simple, and is described as something really complex. Create these actions inside the `store()` method and use list comprehension.